### PR TITLE
feat: add inline examples

### PR DESCRIPTION
**Pull request by pottekkat**
### Description

Adds inline examples to subcommands to improve ease of use.

Now, users can easily refer to the examples to figure out specific flows instead of having to navigate through each option.

The examples are added in an "easy to copy" way with the description of each example in a shell comment form similar to other popular CLIs like `kubectl`. Examples are also added before the `Options` section to make it obvious. Users can look at the docs for specific options after they find the example they want.

Screenshot 2024-08-21 at 12 49 07 PM

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible

### CODE REVIEW

File: apps/cli/src/command/ping.command.ts [hunk 1]
```diff
@@ -10,14 +10,29 @@ export const PingCommand = new BackendCommand<PingOptions>(
...
+      console.log(
+        chalk.green(`Connected to the "${opts.backend}" backend successfully!`),
+      );
+    } catch (err) {
+      console.log(
+        chalk.red(`Unable to connect to the "${opts.backend}" backend. ${err}`),
+      );
+      process.exit(1);
+    }
+  });
```
Comment (pottekkat) [inline apps/cli/src/command/ping.command.ts:38]: Added examples to the ping command and prettify.

**Review by pottekkat — COMMENTED** [review]
To make it easier to review.

File: apps/cli/src/command/sync.command.ts [hunk 2]
```diff
@@ -28,8 +28,36 @@ export const SyncCommand = new BackendCommand<SyncOption>(
...
+      title: 'Synchronize configuration to a specific gateway group',
+      command: 'adc sync -f adc.yaml --gateway-group production',
+    },
+    {
+      title: 'Synchronize configuration without lint check',
+      command: 'adc sync -f adc.yaml --no-lint',
+    },
+    {
+      title: 'Synchronize configuration with debug logs',
+      command: 'adc sync -f adc.yaml --verbose',
```
Comment (bzp2010) [inline apps/cli/src/command/sync.command.ts:50]: I think verbose needs level 2 to print enough debug logs?

File: apps/cli/src/command/helper.ts [hunk 3]
```diff
@@ -32,15 +32,21 @@ export class BaseCommand extends Command {
...
+    const exampleText = this.examples
+      .map((example) => `  # ${example.title}\n  ${example.command}`)
+      .join('\n\n');
+
+    const exampleHeader = this.examples.length === 1 ? 'Example:' : 'Examples:';
+
+    const currDescription = this.description() || '';
+
+    // Append the examples to the description
+    this.description(`${currDescription}\n\n${exampleHeader}\n${exampleText}`);
```
[the remark this answers is not in the document]
Comment (bzp2010) [reply apps/cli/src/command/helper.ts:49]: Can you provide an example of this change? e.g. screenshot or snippet

[the remark this answers is not in the document]
Comment (bzp2010) [reply apps/cli/src/command/helper.ts:49]: Can you provide an example of this change? e.g. screenshot or snippet

File: apps/cli/src/command/sync.command.ts (hunk 2 above)
Comment (pottekkat) [reply apps/cli/src/command/sync.command.ts:50]: Yes. My bad, fixing.

File: apps/cli/src/command/helper.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pottekkat) [reply apps/cli/src/command/helper.ts:49]: Updated the PR description to include the screenshot.

[the remark this answers is not in the document]
Comment (bzp2010) [reply apps/cli/src/command/helper.ts:49]: 🆒

**Review by bzp2010 — APPROVED** [review]
LGTM, UX++

File: apps/cli/src/command/dump.command.ts [hunk 4]
```diff
@@ -74,8 +74,24 @@ export const DumpCommand = new BackendCommand<DumpOptions>(
...
+    {
+      title: 'Save backend configuration to the specified file',
+      command: 'adc dump -o service-configuration.yaml'
+    },
+    {
+      title: 'Save only specified resource types from the backend',
+      command: 'adc dump --include-resource-type global_rule --include-resource-type plugin_metadata',
+    },
+    {
+      title: 'Save only the resources with the specified lables',
```
Comment (kayx23) [inline apps/cli/src/command/dump.command.ts:91]:
```suggestion
      title: 'Save only the resources with the specified labels',
```

File: apps/cli/src/command/diff.command.ts [hunk 5]
```diff
@@ -167,8 +167,16 @@ export const DiffCommand = new BackendCommand<DiffOptions>(
     (filePath, files: Array<string> = []) => files.concat(filePath),
   )
   .addOption(NoLintOption)
-  .addExample('adc diff -f adc.yaml')
-  .addExample('adc diff -f service-a.yaml -f service-b.yaml')
+  .addExamples([
+    {
+      title: 'Compare configuration in specified file with the backend configuration',
```
Comment (kayx23) [inline apps/cli/src/command/diff.command.ts:172]:
```suggestion
      title: 'Compare configuration in a specified file with the backend configuration',
```

**Review by kayx23 — COMMENTED** [review]
others lgtm

**Pull request merged**